pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -607,35 +607,7 @@ func (r *Builder) DataVolumes(vmRef ref.Ref, secret *core.Secret, _ *core.Config
 		// and update the GenerateName field in the DataVolume object.
 		pvcNameTemplate := r.getPVCNameTemplate(vm)
 		if pvcNameTemplate != "" {
-			// Get the VM root disk index
-			planVM := r.getPlanVM(vm)
-			rootDiskIndex := 0
-			if planVM != nil {
-				rootDiskIndex = utils.GetBootDiskNumber(planVM.RootDisk)
-			}
-
-			// Create template data
-			templateData := api.PVCNameTemplateData{
-				VmName:        r.getPlenVMSafeName(vm),
-				PlanName:      r.Plan.Name,
-				DiskIndex:     diskIndex,
-				RootDiskIndex: rootDiskIndex,
-				Shared:        disk.Shared,
-				FileName:      disk.File,
-			}
-
-			generatedName, err := r.executeTemplate(pvcNameTemplate, &templateData)
-			useGenerateName := r.Plan.Spec.PVCNameTemplateUseGenerateName
-
-			// Validate that template output is a valid k8s label
-			errs := k8svalidation.IsDNS1123Label(generatedName)
-			if len(errs) > 0 {
-				// Raise an error if the generated name is not valid
-				err = errors.New("generated PVC name is not valid")
-
-				// Failed to generate PVC name using template
-				r.Log.Info("Failed to generate PVC name using template", "template", pvcNameTemplate, "error", err)
-			}
+			generatedName, useGenerateName := r.GeneratePVCName(vm, disk, diskIndex, pvcNameTemplate)
 
 			if err == nil && generatedName != "" {
```

Comment:
hmm, we don't have `err` as return value from `GeneratePVCName` so it does not help in this `if`

options:
return `err` from `GeneratePVCName`
ensure `generatedName` is "" on err, and remove the test for err == nil

Reply:
was puzzled about that as well